RecoParticleFlow/PFProducer/plugins/PFEGammaProducer.cc

```diff
@@ -104,35 +102,13 @@ PFEGammaProducer::produce(edm::Event& iEvent,
   iSetup.get<ESChannelStatusRcd>().get(esChannelStatusHandle_);
   pfeg_->setESChannelStatus(esChannelStatusHandle_.product());
 
-  // Get The vertices from the event
-  // and assign dynamic vertex parameters
-  edm::Handle<reco::VertexCollection> vertices;
-  bool gotVertices = iEvent.getByToken(vertices_,vertices);
-  if(!gotVertices) {
-    std::ostringstream err;
-    err<<"Cannot find vertices for this event.Continuing Without them ";
-    edm::LogError("PFEGammaProducer")<<err.str()<<std::endl;
-  }
-
   //Assign the PFAlgo Parameters
-  setPFVertexParameters(vertices.product());
```

Comment:
I think if there are no vertices in the event, the call to `Handle::product()` will already raise a readable error I think [1], so the log might not be needed?

[1] https://github.com/cms-sw/cmssw/blob/master/FWCore/Framework/interface/GenericHandle.h#L105